tests/phpunit/test-class-wc-stripe-upe-payment-method.php

```diff
@@ -0,0 +1,319 @@
+<?php
+/**
+ * Unit tests for UPE payment methods
+ */
+class WC_Stripe_UPE_Payment_Method_Test extends WP_UnitTestCase {
+	/**
+	 * Array of mocked UPE payment methods.
+	 *
+	 * @var array
+	 */
+	private $mock_payment_methods = [];
+
+	/**
+	 * Mock capabilities object from Stripe response--all inactive.
+	 */
+	const MOCK_INACTIVE_CAPABILITIES_RESPONSE = [
+		'bancontact_payments' => 'inactive',
+		'card_payments'       => 'inactive',
+		'eps_payments'        => 'inactive',
+		'giropay_payments'    => 'inactive',
+		'ideal_payments'      => 'inactive',
+		'p24_payments'        => 'inactive',
+		'sepa_debit_payments' => 'inactive',
+		'sofort_payments'     => 'inactive',
+		'transfers'           => 'inactive',
+	];
+
+	/**
+	 * Mock capabilities object from Stripe response--all active.
+	 */
+	const MOCK_ACTIVE_CAPABILITIES_RESPONSE = [
+		'bancontact_payments' => 'active',
+		'card_payments'       => 'active',
+		'eps_payments'        => 'active',
+		'giropay_payments'    => 'active',
+		'ideal_payments'      => 'active',
+		'p24_payments'        => 'active',
+		'sepa_debit_payments' => 'active',
+		'sofort_payments'     => 'active',
+		'transfers'           => 'active',
+	];
+
+	/**
+	 * Initial setup
+	 */
+	public function setUp() {
+		parent::setUp();
+		$this->reset_payment_method_mocks();
+	}
+
+	/**
+	 * Reset mock_payment_methods to array of mocked payment methods
+	 * with no mocked expectations for methods.
+	 */
+	private function reset_payment_method_mocks() {
+		$this->mock_payment_methods = [];
+		foreach ( WC_Stripe_UPE_Payment_Gateway::UPE_AVAILABLE_METHODS as $payment_method_class ) {
+			$mocked_payment_method = $this->getMockBuilder( $payment_method_class )
+				->setMethods(
+					[
+						'get_capabilities_response',
+						'get_woocommerce_currency',
+						'is_subscription_item_in_cart',
+					]
+				)
+				->getMock();
+			$this->mock_payment_methods[ $mocked_payment_method->get_id() ] = $mocked_payment_method;
+		}
+	}
+
+	/**
+	 * Helper function to mock subscriptions for internal UPE payment methods.
+	 *
+	 * @param string $function_name Name of function to be mocked.
+	 * @param mixed $value Mocked value for function.
+	 * @param bool $overwrite_mocks Overwrite mocks to remove any existing mocked functions in mock_payment_methods;
+	 */
+	private function set_mock_payment_method_return_value( $function_name, $value, $overwrite_mocks = false ) {
+		if ( $overwrite_mocks ) {
+			$this->reset_payment_method_mocks();
+		}
+
+		foreach ( $this->mock_payment_methods as $mock_payment_method ) {
+			$mock_payment_method->expects( $this->any() )
+				->method( $function_name )
+				->will(
+					$this->returnValue( $value )
+				);
+		}
+	}
+
+	/**
+	 * Convert response array to object.
+	 */
+	private function array_to_object( $array ) {
+		return json_decode( wp_json_encode( $array ) );
+	}
```

Comment:
Not necessarily performant, but performance isn't really required here and this is again the cleanest solution available.